### DISCUSSION TITLE
Does OpenROAD have any plans to support power analysis by reading SAIF files?

### DISCUSSION THREAD

**Discussion by Zhihui-SJTU** [The-OpenROAD-Project/OpenROAD#5773] [Q&A]
I noticed there is an example in the sta/examples/ directory that demonstrates reading a VCD simulation file. I'm wondering if there are any plans to support reading SAIF files, and if so, when it might be available?

**Comment by maliberty** [comment]
There are no plans at this time to support SAIF.  In any case the work would happen in https://github.com/parallaxsw/OpenSTA not this repo.

**Reply by Zhihui-SJTU** [reply]
@maliberty @jjcherry56 @rovinski What are the reasons for not supporting SAIF? It is clear that compared to VCD, SAIF is easier to obtain and results in smaller files. I also noticed that even in the official example, there is a significant difference between power analysis based on VCD and the default settings using set_power_activity -input -activity 0.1 -duty 0.5.

**Reply by rovinski** [reply]
> What are the reasons for not supporting SAIF?

Someone needs to implement it. The OpenROAD team has many more tasks than developers.

**Reply by QuantamHD** [reply]
@Zhihui-SJTU looks like someone has written a parser for it if you want to take a crack at implementing it for your use case.

https://github.com/wsong83/cppSaif

**Reply by Zhihui-SJTU** [reply]
@QuantamHD I also found this repository, but the README is written in Japanese, which is not very user-friendly. So I’m thinking of looking for other parsers. Perhaps you know of any others?

**Reply by Zhihui-SJTU** [reply]
@rovinski 
Got it, hopefully, we can use this feature one day, haha.

**Reply by Zhihui-SJTU** [reply]
@QuantamHD Sorry, please ignore the previous message. I mixed up two repositories.

**Comment by maliberty** [comment]
I think it best to avoid LGPL code.

**Reply by Zhihui-SJTU** [reply]
Do you have any recommended open-source SAIF parsers? Additionally, I would like to ask how much effort it would take if I want to implement my own power analysis support for reading SAIF files. Any suggestions would be greatly appreciated.

**Comment by Zhihui-SJTU** [comment]
@maliberty Here’s some good news: OpenSTA now supports power analysis by reading SAIF files!
